R/RLearner_fcregr_Arima.R

```diff
@@ -0,0 +1,91 @@
+#'@export
+makeRLearner.fcregr.Arima = function() {
+  makeRLearnerForecastRegr(
+    cl = "fcregr.Arima",
+    package = "forecast",
+    par.set = makeParamSet(
+      makeIntegerVectorLearnerParam(id = "order", len = 3L,
+        lower = 0L, upper = Inf,
+        default = c(0L,0L,0L)),
+      makeIntegerVectorLearnerParam(id = "seasonal", len = 3L,
+        lower = 0L, upper = Inf,
+        default = c(0L,0L,0L)),
+      makeLogicalLearnerParam(id = "include.mean", default = TRUE),
+      makeLogicalLearnerParam(id = "include.drift", default = FALSE),
+      makeNumericLearnerParam(id = "lambda", default = NULL, special.vals = list(NULL), when = "both"),
+      makeLogicalLearnerParam(id = "biasadj", default = FALSE, when = "both"),
+      makeDiscreteLearnerParam(id = "method", values = c("CSS-ML", "ML", "CSS"), default = "CSS-ML"),
+      makeUntypedLearnerParam(id = "model", default = NULL),
+      # arima params
+      makeLogicalLearnerParam(id = "transform.pars", default = TRUE),
+      makeNumericVectorLearnerParam(id = "fixed", len = NA, default = NULL, special.vals = list(NULL)),
+      makeNumericVectorLearnerParam(id = "init", len = NA, default = NULL, special.vals = list(NULL)),
+      # No default
+      makeIntegerLearnerParam("n.cond", lower = 0),
+      makeDiscreteLearnerParam("SSinit", values = c("Gardner1980", "Rossignol2011"), default = "Gardner1980", tunable = FALSE),
+      makeDiscreteLearnerParam("optim.method", default = "BFGS", values = c("Nelder-Mead", "BFGS",
+        "CG", "L-BFGS-B",
+        "SANN", "Brent"),
+                               tunable = FALSE),
+      makeUntypedLearnerParam("optim.controls", default = list(), tunable = FALSE),
+      makeNumericLearnerParam("kappa", lower = 1e6, upper = Inf, tunable = FALSE),
+      # prediction params
+      makeIntegerLearnerParam(id = "h", lower = 0, upper = Inf,
+        default = expression(ifelse(object$arma[5]>1,2*object$arma[5],10)),
+        tunable = FALSE,
```

Comment:
This is for the forecasting length. So whether to forecast 1 or 2 or 100 periods in to the future. Naturally if we tune this we would find that the minimum forecast error comes from forecasting 1 period into the future (or the minimum selected). So I don't really see this as something that should be tuned, but sure I can allow it